lib/maxmind/db/decoder.rb

```diff
@@ -0,0 +1,246 @@
+require 'maxmind/db/errors'
+
+module MaxMind # :nodoc:
+  class DB
+    # +Decoder+ decodes a {MaxMind DB}[http://maxmind.github.io/MaxMind-DB/]
+    # data section.
+    #
+    # Typically you will interact with this class through a Reader rather than
+    # directly.
+    class Decoder # :nodoc:
+      # Create a +Decoder+.
+      #
+      # +io+ is the DB. It must provide a +read+ method. It must be opened in
+      # binary mode.
+      #
+      # +pointer_base+ is the base number to use when decoding a pointer. It is
+      # where the data section begins rather than the beginning of the file.
+      # The specification states the formula in the `Data Section Separator'
```

Comment:
Yes, because I am a hipster https://english.stackexchange.com/questions/17695/any-reference-on-the-usage-of-a-backtick-and-single-quotation-mark-like-this